source/workspaced/com/dcd.d

```diff
@@ -228,15 +228,32 @@ class DCDComponent : ComponentWrapper
 		serverThreads.create({
 			mixin(traceTask);
 			if (quietServer)
+			{
 				foreach (block; serverPipes.stderr.byChunk(4096))
 				{
 				}
+			}
 			else
-				while (serverPipes.stderr.isOpen && !serverPipes.stderr.eof)
+			{
+				try
+				{
+					while (running)
+					{
+						auto line = serverPipes.stderr.readln();
+						if (line !is null)
+							trace("Server: ", line); // evaluates lazily, so read before
+						else
```

Comment:
hm is this guaranteed to work though? The phobos documentation says nothing about readln ever returning null or it meaning it's no longer running.

I think readln definitely returns nothing when the File is eof or no longer opened, which the previous while checked.

Reply:
io errors should be handled via the exception it throws anyways, the ``!is null`` follows the example in the doc..

It's a pretty complex bug to track..